packages/contracts/contracts/core/SphinxGnosisSafeProxyFactory.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+import { ISphinxGnosisSafeProxyFactory } from "./interfaces/ISphinxGnosisSafeProxyFactory.sol";
+import { ISphinxModuleProxyFactory } from "./interfaces/ISphinxModuleProxyFactory.sol";
+import { GnosisSafeVersion } from "./SphinxDataTypes.sol";
+// Import Gnosis Safe proxy v1.3.0 and v1.4.1, which we'll deploy in this file.
+import {
+    GnosisSafeProxy as GnosisSafeProxy_v1_3_0
+} from "@gnosis.pm/safe-contracts-1.3.0/proxies/GnosisSafeProxy.sol";
+import {
+    SafeProxy as GnosisSafeProxy_v1_4_1
+} from "@gnosis.pm/safe-contracts-1.4.1/proxies/SafeProxy.sol";
+// Import contracts from Gnosis Safe v1.3.0, which we'll use as interfaces in this file. All of
+// these contracts have the same interface as Gnosis Safe v1.4.1 for the functions used in this
+// contract.
+import { MultiSend } from "@gnosis.pm/safe-contracts-1.3.0/libraries/MultiSend.sol";
+import { GnosisSafe } from "@gnosis.pm/safe-contracts-1.3.0/GnosisSafe.sol";
+import { Enum } from "@gnosis.pm/safe-contracts-1.3.0/common/Enum.sol";
+
+contract SphinxGnosisSafeProxyFactory is ISphinxGnosisSafeProxyFactory {
+    ISphinxModuleProxyFactory private immutable moduleProxyFactory;
+
+    MultiSend private immutable multiSend;
+
+    constructor(ISphinxModuleProxyFactory _moduleProxyFactory, MultiSend _multiSend) {
+        moduleProxyFactory = _moduleProxyFactory;
+        multiSend = _multiSend;
+    }
+
+    /**
+     * @param _owners The addresses that will own the deployed Gnosis Safe. These should be sorted
+     *                in ascending order off-chain so that the address of the Gnosis Safe can easily
+     *                be calculated by off-chain tooling.
+     */
+    function deployGnosisSafeWithModule(
```

Comment:
This function includes a lot of logic that was previously off-chain. Specifically, it contains all of the logic for constructing the Gnosis Safe's initializer data, which involves a MultiSend call. Although including this logic in the contract may not be strictly necessary, I included it for two reasons:
* I think this logic is important enough to include in the audit
* Putting this logic on-chain makes it easier for users to inspect the transaction that deploys their Gnosis Safe